torchhd/datasets/abalone.py

```diff
@@ -0,0 +1,165 @@
+
+        if self.target_transform:
+            label = self.target_transform(label)
+
+        return sample, label
+
+    def _check_integrity(self) -> bool:
+        if not os.path.isdir(self.root):
+            return False
+
+        # Check if the root directory contains the required files
+        has_train_file = os.path.isfile(os.path.join(self.root, "abalone_R.dat"))
+        has_k_fold_file = os.path.isfile(os.path.join(self.root, "conxuntos_kfold.dat"))
+        if has_train_file and has_k_fold_file:
+            return True
+
+        # TODO: Add more specific checks like an MD5 checksum
+
+        return False
+
+    def _load_data(self):
+        train_data_file = "abalone_R.dat"
+        val_file = "conxuntos_kfold.dat"
+
+        if self.train:
+            train_data = pd.read_csv(
+                os.path.join(self.root, train_data_file),
+                sep="\t",
+                header=None,
+                skiprows=1,
+            )
+            train_data_all = train_data.values[:, 1:-1]
+            train_targets_all = train_data.values[:, -1].astype(int)
+
+            if self.fold_id == 0:
+                self.data = torch.tensor(train_data_all, dtype=torch.float)
+                self.targets = torch.tensor(train_targets_all, dtype=torch.long)
+            else:
+                cross_val_ind = pd.read_csv(
+                    os.path.join(self.root, val_file), sep=" ", header=None
+                )
+                cross_val_ind = cross_val_ind.values[:, 0:-1]
+                fold_index = (self.fold_id - 1) * 2 + int(self.fold_val)
+                k_fold_indices = cross_val_ind[
+                    fold_index, np.invert(np.isnan(cross_val_ind[fold_index, :]))
+                ].astype(int)
+                self.data = torch.tensor(
+                    train_data_all[k_fold_indices, :], dtype=torch.float
+                )
+                self.targets = torch.tensor(
+                    train_targets_all[k_fold_indices], dtype=torch.long
+                )
+
+        else:
+            raise ValueError(
+                f"This dataset does not have a separate file for test data."
+            )
+
+    def download(self):
+        """Download the data if it doesn't exist already."""
+
+        if self._check_integrity():
+            print("Files are already downloaded and verified")
+            return
+
+        url = "http://persoal.citius.usc.es/manuel.fernandez.delgado/papers/jmlr/data.tar.gz"
+
+        data_dir = os.path.join(self.root, os.pardir)
+        archive_path = os.path.join(data_dir, "data_hundreds_classifiers.tar.gz")
+
+        if os.path.isfile(archive_path):
+            print("Archive file is already downloaded")
+        else:
+            download_file(url, archive_path)
+
+        # Extract archive
+        with tarfile.open(archive_path) as file:
+            for member in file.getmembers():
+                if member.name.startswith("abalone"):
```

Comment:
I made it such that it only extracts the relevant files from the archive which was significantly faster than unpacking everything I found.

Reply:
Thanks, this is neat!